File: clients_java/client/src/org/openamq/client/transport/AMQProtocolProvider.java
package org.openamq.client.transport;

import org.apache.mina.filter.codec.ProtocolCodecFactory;
import org.apache.mina.filter.codec.demux.DemuxingProtocolCodecFactory;
import org.apache.mina.handler.demux.DemuxingIoHandler;
import org.openamq.framing.AMQDataBlockDecoder;
import org.openamq.framing.AMQDataBlockEncoder;
import org.openamq.framing.ProtocolInitiation;

/**
 * Creates the codec factory and registers the codecs that are used to encode
 * (and decode) data sent to and from the server. All the code
 */
public class AMQProtocolProvider extends DemuxingIoHandler
{
    private DemuxingProtocolCodecFactory _factory;

    public AMQProtocolProvider()
    {
        _factory = new DemuxingProtocolCodecFactory();
        _factory.register(AMQDataBlockEncoder.class);

        AMQDataBlockDecoder decoder = new AMQDataBlockDecoder();
        _factory.register(decoder);
        _factory.register(new ProtocolInitiation.Decoder());
    }

    public ProtocolCodecFactory getCodecFactory()
    {
        return _factory;
    }
}

File: dotnet/xmsclient/Client/Transport/AMQProtocolProvider.cs
using System;
using jpmorgan.mina.common;
using jpmorgan.mina.filter.codec;
using jpmorgan.mina.filter.codec.demux;
using jpmorgan.mina.handler.demux;
using log4net;
using OpenAMQ.Framing;

namespace OpenAMQ.XMS.Client.Transport
{
    public class AMQProtocolProvider : DemuxingIoHandler
    {
        private DemuxingProtocolCodecFactory _factory;
        
        public AMQProtocolProvider()
        {
            _factory = new DemuxingProtocolCodecFactory();
            _factory.Register(typeof (AMQDataBlockEncoder));
            _factory.Register(new ProtocolInitiation.Decoder());
        }
        
        public IProtocolCodecFactory CodecFactory
        {
            get
            {
                return _factory;
            }
        }
    }
}
